src/sage/schemes/elliptic_curves/ell_field.py

```diff
@@ -1636,6 +1636,65 @@ def isogenies_prime_degree(self, l=None, max_l=31):
         from .isogeny_small_degree import isogenies_prime_degree
         return sum([isogenies_prime_degree(self, d) for d in L], [])
 
+    def isogenies_degree(self, n):
+        r"""
+        Return a list of all separable isogenies of given degree with domain
+        equal to ``self``, which are defined over the base field of ``self``.
+
+        INPUT:
+
+        - ``n`` -- an integer.
+
+        TESTS::
+
+            sage: E = EllipticCurve(GF(11), [1, 1])
+            sage: E.isogenies_degree(23 * 19)
+            [Composite morphism of degree 437 = 1*19*23:
+               From: Elliptic Curve defined by y^2 = x^3 + x + 1 over Finite Field of size 11
+               To:   Elliptic Curve defined by y^2 = x^3 + 8*x + 7 over Finite Field of size 11,
+             Composite morphism of degree 437 = 1*19*23:
+               From: Elliptic Curve defined by y^2 = x^3 + x + 1 over Finite Field of size 11
+               To:   Elliptic Curve defined by y^2 = x^3 + 2*x + 6 over Finite Field of size 11,
+             Composite morphism of degree 437 = 1*19*23:
+               From: Elliptic Curve defined by y^2 = x^3 + x + 1 over Finite Field of size 11
+               To:   Elliptic Curve defined by y^2 = x^3 + 6*x + 2 over Finite Field of size 11,
+             Composite morphism of degree 437 = 1*19*23:
+               From: Elliptic Curve defined by y^2 = x^3 + x + 1 over Finite Field of size 11
+               To:   Elliptic Curve defined by y^2 = x^3 + 7*x + 8 over Finite Field of size 11]
+
+        ::
+
+            sage: pol = PolynomialRing(QQ, 'x')([1, -3, 5, -5, 5, -3, 1])
+            sage: L.<a> = NumberField(pol)
+            sage: js = hilbert_class_polynomial(-23).roots(L, multiplicities=False)
+            sage: len(js)
+            3
+            sage: E = EllipticCurve(j=js[0])
+            sage: len(E.isogenies_degree(2**2))
+            7
+            sage: len(E.isogenies_degree(2**5)) # long time (15s)
+            99
+        """
+        n = Integer(n)
+        if n.is_prime():
+            return self.isogenies_prime_degree(n)
+
+        prime_divisors = sum([[p] * e for p, e in n.factor()], [])
+        isos = [self.isogeny(self(0))]
```

Comment:
Here I would use `sage.schemes.elliptic_curves.weierstrass_morphism.identity_morphism()` — The difference is that this one will be an explicit factor in the resulting `EllipticCurveHom_composite` whereas the result of `identity_morphism()` won't, so this is a little bit cleaner.